feature/dynamodb/attributevalue/decode.go

```diff
@@ -212,6 +212,16 @@ type DecoderOptions struct {
 	// Number type instead of float64 when the destination type
 	// is interface{}. Similar to encoding/json.Number
 	UseNumber bool
+
+	// Will decode S attribute values and SS attribute value elements into time.Time
+	//
+	// Default string parsing format is time.RFC3339
+	DecodeTimeS func(string) (time.Time, error)
+
+	// Will decode N attribute values and NS attribute value elements into time.Time
+	//
+	// Default number parsing format is seconds since January 1, 1970 UTC
+	DecodeTimeN func(string) (time.Time, error)
```

Comment:
What are your thoughts about putting these on a separate structure to group them for discoverability?

```go
type DecodeTimeAttributes struct {
    S func(string) (time.Time, error)
    N func(string) (time.Time, error)
}
```

This would allow for easy extension for adding `SS` and `NS` attributes down the road if we want to, but without having to have an explosion of top-level fields on the `DecoderOptions` structure. As on the top level structure we would have `DecodeTime DecodeTimeAttributes`.

Reply:
done